test/lib/set-script.js

```diff
@@ -1,7 +1,38 @@
 const test = require('tap')
 const requireInject = require('require-inject')
 const parseJSON = require('json-parse-even-better-errors')
+const { writeFileSync } = require('fs')
 
+test.test('completion', t => {
+  const SetScript = requireInject('../../lib/set-script.js')
+  const dir = t.testdir()
+  const setScript = new SetScript({ localPrefix: dir })
+  t.test('already have a script name', async t => {
+    const res = await setScript.completion({conf: {argv: {remain: ['npm', 'run', 'x']}}})
+    t.equal(res, undefined)
+    t.end()
+  })
+  t.test('no package.json', async t => {
+    const res = await setScript.completion({conf: {argv: {remain: ['npm', 'run']}}})
+    t.strictSame(res, [])
+    t.end()
+  })
+  t.test('has package.json, no scripts', async t => {
+    writeFileSync(`${dir}/package.json`, JSON.stringify({}))
+    const res = await setScript.completion({conf: {argv: {remain: ['npm', 'run']}}})
+    t.strictSame(res, [])
+    t.end()
+  })
+  t.test('has package.json, with scripts', async t => {
+    writeFileSync(`${dir}/package.json`, JSON.stringify({
+      scripts: { hello: 'echo hello', world: 'echo world' },
+    }))
```

Comment:
better to just use **tap** fixtures helper here instead:

```suggestion
  const localPrefix = t.testdir({
    'package.json': JSON.stringify({
      scripts: { hello: 'echo hello', world: 'echo world' },
    })
  })
  const setScript = new SetScript({ localPrefix })
    
```